modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/components/LayoutReports.js

```diff
@@ -13,82 +13,117 @@
  */
 
 import ClayAlert from '@clayui/alert';
-import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayButton from '@clayui/button';
+import ClayProgressBar from '@clayui/progress-bar';
 import {useIsMounted} from '@liferay/frontend-js-react-web';
 import {fetch} from 'frontend-js-web';
 import PropTypes from 'prop-types';
-import React, {useCallback, useEffect, useReducer} from 'react';
+import React, {useCallback, useContext, useEffect, useState} from 'react';
 
+import {StoreDispatchContext, StoreStateContext} from '../context/StoreContext';
+import APIService from '../utils/APIService';
 import BasicInformation from './BasicInformation';
 import EmptyLayoutReports from './EmptyLayoutReports';
-
-const initialState = {
-	data: null,
-	error: null,
-	loading: false,
-};
-
-const dataReducer = (state, action) => {
-	switch (action.type) {
-		case 'LOAD_DATA':
-			return {
-				...state,
-				loading: true,
-			};
-
-		case 'SET_ERROR':
-			return {
-				...state,
-				error: action.error,
-				loading: false,
-			};
-
-		case 'SET_DATA':
-			return {
-				data: {
-					...action.data,
-				},
-				error: action.data?.error,
-				loading: false,
-			};
-
-		default:
-			return initialState;
-	}
-};
+import LayoutReportsIssuesList from './LayoutReportsIssuesList';
 
 export default function LayoutReports({
 	eventTriggered,
 	isPanelStateOpen,
 	layoutReportsDataURL,
+	portletNamespace,
 }) {
 	const isMounted = useIsMounted();
 
-	const [state, dispatch] = useReducer(dataReducer, initialState);
+	const {data, error, loading} = useContext(StoreStateContext);
+
+	const dispatch = useContext(StoreDispatchContext);
 
 	const safeDispatch = useCallback(
 		(action) => {
 			if (isMounted()) {
 				dispatch(action);
 			}
 		},
-		[isMounted]
+		[dispatch, isMounted]
 	);
 
+	const [percentage, setPercentage] = useState(0);
+
+	useEffect(() => {
+		if (loading && !error) {
+			const x = Date.now();
+			const interval = setInterval(() => {
+				const time = (Date.now() - x) / 1000;
+				const progress =
+					-200 * (1 / (1 + Math.pow(Math.E, time / 4)) - 0.5);
```

Comment:
I didn't even know there was a `Math.E` constant. I'm not going to pretend to understand this, so...

![proceed](https://user-images.githubusercontent.com/7074/116525913-4f7e9f00-a8d9-11eb-8480-b4f3128e110d.gif)

Reply:
This is a calculation provided by @4lejandrito to calculate from 0 to ~100 (never gets to 100) and this is used to update the progress bar value because when we call Google PageSpeed endpoint we don't have any information about the progress so... Sorry? 😬

Reply:
It's [a sigmoid function](https://en.wikipedia.org/wiki/Sigmoid_function):

* Scaled to go from [0, 100). It goes up smoothly and never gets to 100. 
* It's adjusted so that it gets close to 100 in around 15 seconds (the time it normally takes to complete the request to google pagespeed)

We need to agree on a better timing and I will send that in a follow up.

Reply:
I will also make the parameters (15 seconds and 100 more explicit using variables).